indexing-service/src/main/java/org/apache/druid/indexing/input/DruidInputSource.java

```diff
@@ -313,6 +326,10 @@ private List<TimelineObjectHolder<String, DataSegment>> createTimeline()
     if (interval == null) {
       return getTimelineForSegmentIds(coordinatorClient, dataSource, segmentIds);
     } else {
+      if (toolbox != null) {
```

Comment:
In which cases will this be `null` here?

Reply:
It can be null when it is being used to sample data before ingestion begins